Git.hub/Client.cs

```diff
@@ -99,33 +94,59 @@ public IList<Repository> getRepositories(string username)
         /// <returns>fetched repository</returns>
         public Repository getRepository(string username, string repositoryName)
         {
-            var request = new RestRequest("/repos/{name}/{repo}");
-            request.AddUrlSegment("name", username);
-            request.AddUrlSegment("repo", repositoryName);
+            var request = new RestRequest("/repos/{name}/{repo}")
+                .AddUrlSegment("name", username)
+                .AddUrlSegment("repo", repositoryName);
 
-            var repo = client.Get<Repository>(request).Data;
+            var repo = DoRequest<Repository>(request);
             if (repo == null)
                 return null;
 
-            repo._client = client;
+            repo._client = _client;
             repo.Detailed = true;
             return repo;
         }
 
+        private T DoRequest<T>(IRestRequest request, bool shouldNotFail = false) where T : new()
```

Comment:
Please invert `shouldNotFail`. It leads to a mental overload when staring at `true` to understand what it means.
```suggestion
        private T DoRequest<T>(IRestRequest request, bool throwOnError = true) where T : new()
```